crux-core/src/crux/query.clj

```diff
@@ -1183,11 +1183,18 @@
   (newSnapshot [this]
     (lru/new-cached-snapshot (kv/new-snapshot (:kv this)) true))
 
+  ;; TODO query-hooks here
   (q [this q]
-    (crux.query/q this q))
+    (let [latter-fns (seq (map #(% {:query q}) query-hooks))
+          result (crux.query/q this q)]
+      (seq (map #(% {:result result :query q}) latter-fns))
```

Comment:
reckon you're using `seq` here with the intention of forcing the sequence? (like the Haskell function, I think?)

in Clojure, `seq` only happens to force the _first_ element of a sequence (hence why your test passes) - you'd want something like `doseq`, `dorun`, or, better, just `run!` here (given you're only after the side effects)

(the one above, L1188, can remain lazy)

Reply:
`seq` does force the whole sequence, but I agree another function is better suited here